src/js/components/List.jsx

```diff
@@ -0,0 +1,197 @@
+    style?: Object,
+    data?: Array<Object>,
+    labelFunction?: Function,
+    defaultStyles: bool,
+    defaultContainerClass: bool,
+    defaultItemClass: bool,
+    defaultSelection?: Object,
+    onItemSelect?: Function,
+}
+
+type State = {
+    selectedItem?: Object,
+};
+
+// TODO: fix a few flow errors
+/*
+type RendererProps = {
+    listIndex: number,
+    listItem: string | Object,
+    labelFunction: Function,
+}
+*/
+
+
+export class List extends React.Component {
+
+    props: Props;
+    state: State;
+
+    static defaultProps: Props = {
+        style: {},
+        defaultStyles: true,
+        defaultContainerClass: true,
+        defaultItemClass: true,
+    };
+
+    constructor(props:Props) {
+        super(props);
+
+        this.state = {
+            selectedItem: null,
+        };
+    }
+
+    componentWillReceiveProps(nextProps:Props) {
+        if (this.props.data !== nextProps.data) {
+            this.setState({
+                selectedItem: null,
+            });
+        }
+
+        this._defaultSelection(nextProps);
+    }
+
+    componentDidMount() {
+        this._defaultSelection(this.props);
+    }
+
+    get selectedIndex():number {
+        return this.props.data ? this.props.data.indexOf(this.state.selectedItem) : -1;
+    }
+
+    get selectedItem():Object {
+        return this.state.selectedItem;
+    }
+
+    _defaultSelection(nextProps:Props) {
+        if (!this.state.selectedItem && nextProps.defaultSelection) {
+            this.setState({
+                selectedItem: nextProps.defaultSelection,
+            });
+        }
+    }
+
+    _onClickListItem(event:Event, index:number, item:Object) {
+        event.preventDefault();
+
+        this._selectItem(index, item);
+    }
+
+    _onKeyPressListItem(event:React$Event, index:number, item:Object) {
+        event.preventDefault();
+
+        if (event.nativeEvent.keyCode === KeyCodes.SPACEBAR) {
+            this._selectItem(index, item);
+        }
+    }
+
+    _selectItem(index:number, item:Object) {
+        this.setState({
+            selectedItem: item
+        });
+
+        if (this.props.onItemSelect) {
+            this.props.onItemSelect(index, item);
+        }
+    }
+
+    render() {
+        const { children, data, keyFunction, labelFunction } = this.props;
+
+        const childCount = React.Children.count(children);
+
+        let childTemplate = null;
+
+        if (childCount === 0) {
+            childTemplate = <DefaultRenderer />;
+        } else if (childCount === 1) {
+            childTemplate = React.Children.only(children);
+        } else {
+            console.error(`'children' supplied to List must be zero or one elements`);
+            return null;
+        }
+
+        const listClass = this.props.className || '';
+        const selectedClass = this.state.selectedItem ? 'List-selected' : '';
+
+        let containerClass = '', itemClass = '';
+
+        if (this.props.defaultStyles) {
```

Comment:
Not sure I agree with the default styling here. I'd rather have a functional shell of a list that could have a top-level class applied and style it in some different, common ways.